webdriver/firefox.go

```diff
@@ -0,0 +1,80 @@
+package webdriver
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/tebeka/selenium"
+	"github.com/tebeka/selenium/firefox"
+)
+
+var (
+	seleniumPath     = flag.String("selenium_path", "", "Path to the selenium standalone binary.")
+	geckoDriverPath  = flag.String("geckodriver_path", "", "Path to the geckodriver binary")
+	firefoxPath      = flag.String("firefox_path", "", "Path to the firefox binary")
+	startFrameBuffer = flag.Bool("frame_buffer", frameBufferDefault(), "Whether to use a frame buffer")
+	seleniumHost     = flag.String("selenium_host", "localhost", "Host to run selenium on")
+	seleniumPort     = flag.Int("selenium_port", 8888, "Port to run selenium on")
+)
+
+func frameBufferDefault() bool {
+	return runtime.GOOS != "darwin"
+}
+
+// FirefoxWebDriver starts up a Firefox WebDriver.
+// Make sure to close both the service and the WebDriver instances, e.g.
+//
+// s, d, e := FirefoxWebDriver()
+// if e != nil {
+//   panic(e)
+// }
+// defer s.Stop()
+// defer wd.Quit()
```

Comment:
This is probably just my lack of context/knowledge in go, but where does this wd variable come from in the example? I assume wd is webdriver, but from where? If its the selenium WebDriver instance does it make sense to also include that definition in the example here?

Also example comments like these are awesome :)

Reply:
copy pasta, fixed!